### PR TITLE
Fix/slash commands bugs

### LINKED ISSUE #5741: [Bug] [intermittent] Drag and drop the widget and  click on slash icon  all the data sources suggestion are not displayed in the dropdown
## Description

[What happened]when the first time user drag and drop the widget and  click on slash icon  all the data sources suggestion are not displayed in the dropdown

### Steps to reproduce the behaviour:

https://drive.google.com/file/d/18YDJywyBGVKVPjyxE4SLg1-Asj09zpr3/view

### Important Details

- Version: [Cloud ]
- OS: Windows
- Browser Chrome
- Environment  release

### LINKED ISSUE #5742: [Bug]The slash command dropdown opens up in a random place away from the widget and property pane
## Description

[What happened ]The slash command dropdown opens up in a random place away from the widget and property pane

### Steps to reproduce the behaviour:

1.drag and drop the widget
2.Inside the property pane 
3.click on slash icon
4.The dropdown is displayed far away from the property pane

image

### Important Details

- Version: [Cloud]
- OS: Windows
- Browser Chrome
- Environment release

### LINKED ISSUE #5825: [Bug] Slash Command Issues
- [x] When the menu is open, the blue tooltip behind it should not show
- [x] The headers of the menu should look the same as the headers of the autocomplete
- [x] New Datasource does not have an icon
- [x] Pressing enter on new binding does not open the autocomplete but clicking it does
- [x] As soon as you focus on the field with slash commands, if you click an option it does not work only enter works
- [x] Bind data in the query does not populate the field as well. This should work exactly like the best match feature
- [x] Slashmenu does not work in the keys of the API pane and the tooltip for the key field is different from the value field in the headers

### PR DESCRIPTION
**Pull request by arunvjn**
## Description
Fixes a lot of minor bugs around slash commands

Fixes #5825 
Fixes #5742 
Fixes #5741 

## Type of change
- Bug fix (non-breaking change which fixes an issue)

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

## Test coverage results :test_tube:
<details><summary>:red_circle: Total coverage has decreased</summary>

    // Code coverage diff between base branch:release and head branch: fix/slash_commands_bugs 
Status | File | % Stmts | % Branch | % Funcs | % Lines 
 -----|-----|---------|----------|---------|------ 
 :red_circle: | total | 53.59 **(-0.06)** | 35.37 **(-0.08)** | 32.1 **(-0.01)** | 54.15 **(-0.05)**
 :red_circle: | app/client/src/components/editorComponents/CodeEditor/commandsHelper.ts | 16.95 **(-25.16)** | 0 **(-40)** | 10 **(-12.22)** | 16.67 **(-27.56)**
 :green_circle: | app/client/src/components/editorComponents/CodeEditor/generateQuickCommands.tsx | 23.44 **(6.2)** | 7.14 **(7.14)** | 6.67 **(6.67)** | 22.22 **(6.43)**
 :red_circle: | app/client/src/components/editorComponents/CodeEditor/index.tsx | 69.1 **(-2.02)** | 43.28 **(-4.05)** | 50 **(-2.63)** | 70.18 **(-1.51)**
 :red_circle: | app/client/src/components/formControls/DynamicTextFieldControl.tsx | 98.11 **(-0.04)** | 70.59 **(0)** | 88.89 **(0)** | 98 **(-0.04)**</details>

### CODE REVIEW

File: app/client/src/components/editorComponents/CodeEditor/generateQuickCommands.tsx [hunk 1]
```diff
@@ -25,27 +31,27 @@ export const generateQuickCommands = (
...
     text: "{{}}",
     displayText: "New Binding",
-    shortcut: "{{}}",
+    shortcut: Shortcuts.BINDING,
   });
-  const newDatasource: CommandsCompletion = generateCreateNewCommand({
+  const newIntegration: CommandsCompletion = generateCreateNewCommand({
     text: "",
-    displayText: "New Datasource",
+    displayText: "New Integration",
```
Comment (Nikhil-Nandagopal) [inline app/client/src/components/editorComponents/CodeEditor/generateQuickCommands.tsx:42]: @arunvjn this should be New Datasource

File: app/client/src/components/editorComponents/CodeEditor/commandsHelper.ts [hunk 2]
```diff
@@ -81,16 +75,18 @@ export const commandsHelper: HintHelper = (editor, data: any) => {
...
-                0,
-                value.length - searchText.length - 1,
-              );
-              if (selected.action && typeof selected.action === "function") {
-                updatePropertyValue(updatedValue, updatedValue.length, true);
-                selected.action();
-              } else {
-                updatePropertyValue(updatedValue + selected.text);
-              }
+              setTimeout(() => {
```
Comment (ApekshaBhosale) [inline app/client/src/components/editorComponents/CodeEditor/commandsHelper.ts:78]: why setTimeout needed here?

**Review by ApekshaBhosale — CHANGES_REQUESTED** [review]

File: app/client/src/components/editorComponents/CodeEditor/commandsHelper.ts (hunk 2 above)
Comment (arunvjn) [reply app/client/src/components/editorComponents/CodeEditor/commandsHelper.ts:78]: We need a delay for the editor's value to update before changing the position of the cursor and execute the subsequent action.

**Comment by ApekshaBhosale** [comment]
Approving for deploy review for now @arunvjn

**Review by ApekshaBhosale — DISMISSED** [review]

File: app/client/src/components/editorComponents/CodeEditor/commandsHelper.ts (hunk 2 above)
Comment (arunvjn) [reply app/client/src/components/editorComponents/CodeEditor/commandsHelper.ts]: Removed dependency on timeout

**Review by ApekshaBhosale — DISMISSED** [review]

File: app/client/src/utils/autocomplete/TernServer.ts [hunk 3]
```diff
@@ -287,12 +287,14 @@ class TernServer {
...
-      ...otherCompletions,
-    ];
+    return Array.from(
+      new Set([
+        ...docCompletetions,
+        ...sameDataType,
+        ...otherDataType,
+        ...otherCompletions,
+      ]),
+    );
```
Comment (hetunandu) [inline app/client/src/utils/autocomplete/TernServer.ts:297]: Remove this change

**Review by ApekshaBhosale — APPROVED** [review]

**Review by ApekshaBhosale — APPROVED** [review]

**Pull request merged**